### ISSUE TITLE
Proposal: Do not show animations when switching menus in the MenuBar

### ISSUE THREAD

**Issue by Poopooracoocoo** [microsoft/microsoft-ui-xaml#2517]
# Proposal: Do not show animations when switching menus in the MenuBar

## Summary

Do not show animations when switching menus in the MenuBar. 

## Rationale

* Feels faster
* Consistency with whatever notepad uses

## Scope

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to accomplish W | Must |
| This proposal will allow end users to accomplish X | Should |
| This proposal will allow developers to accomplish Y | Could |
| This proposal will allow end users to accomplish Z | Won't |

**Comment by BreeceW** [comment]
The animations on MenuBar are generally slow for what I would consider to be a rather legacy control (in terms of design) where a user might expect an instant response instead of a sliding animation. The animation also looks a little comical for very long menus.

**Comment by Poopooracoocoo** [comment]
@shaheedmalik @electronic-dk 
https://gofile.io/d/3ohawG

**Comment by Poopooracoocoo** [comment]
@BreeceW yeah many of the animations are superfluous. I'm pretty sure I've suggested toning them down for longer flyouts and menus somewhere 'round here.

**Comment by electronic-dk** [comment]
I wouldn't say I dislike these animations. Perhaps they may be sped up a bit, but I'm not sure about it. There's already an OS-wide setting for showing animations, does it not cover this?

**Comment by Poopooracoocoo** [comment]
@electronic-dk I'm only talking about the animations when you're switching menus, not the first slide in animation. win32 apps don't have animations when you're switching menus either. 

> There's already an OS-wide setting for showing animations, does it not cover this?

In case you're actually asking, yes it does but since we both understand each other, defaults freaking matter. Repeating a slide in animation feels unnecessary and slow and as we both know, win32 apps still have a fade in animation (which you can disable) for menus. 

tbh this whole discussion is kinda pointless as idk if any of my suggestions have done anything. windows for example hasn't fixed the reveal hover bug in an entire year, let alone done any of the bigger things.

**Comment by mdtauk** [comment]
Animated UI is a modern thing, and will become more expected from users coming from their Android or Apple phones.

**Comment by Poopooracoocoo** [comment]
@mdtauk I think you misunderstood what I'm saying. I'm not against animated UI and this isn't a modern control as Breece said.

**Comment by mdtauk** [comment]
> @mdtauk I think you misunderstood what I'm saying. I'm not against animated UI and this isn't a modern control as Breece said.

It is a modern version of a classic UI 😀

**Comment by contextfree** [comment]
I definitely agree that the experience and efficiency of browsing menus by "scrubbing" across them is made significantly worse by having it repetitively slide up and down as you're doing it. 

This isn't about whether UI animations in general are generically good or bad (and it doesn't really depend on whether it's a "legacy" control or not, though it's worthwhile considering the target audience and scenarios) Animations can make UI more pleasant and useful (by letting user viscerally feel the application's structure), but they can be tricky to get right, and inappropriate use of animations is worse than no animations. If you were designing an animation for scrubbing along menus it wouldn't look like the menu sliding up and down repeatedly. It might look more like how thumbnails on the Windows taskbar scoot along and crossfade/morph into each other as you scrub along it, though that one's still slower than I'd like here.

**Comment by electronic-dk** [comment]
IMHO, the biggest issue here is not the animation or no animation per se, but the fact the menus in win32 and UWP apps look (and behave) differently. With that being said I'd prefer if all the menus/context menus were modernized and gained a common look and behavior (e.g. animation/no animation).

**Comment by Poopooracoocoo** [comment]
> IMHO, the biggest issue here is not the animation or no animation per se, but the fact the menus in win32 and UWP apps look (and behave) differently. With that being said I'd prefer if all the menus/context menus were modernized and gained a common look and behavior (e.g. animation/no animation).

yeah. windows 10 is an inconsistent mess and microsoft doesn't want to even fix it :( there are options in control panel that don't affect WinUI stuff like shadows for example

**Comment by mdtauk** [comment]
> > IMHO, the biggest issue here is not the animation or no animation per se, but the fact the menus in win32 and UWP apps look (and behave) differently. With that being said I'd prefer if all the menus/context menus were modernized and gained a common look and behavior (e.g. animation/no animation).
> 
> yeah. windows 10 is an inconsistent mess and microsoft doesn't want to even fix it :( there are options in control panel that don't affect WinUI stuff like shadows for example

- Control Panel is due to be removed, once the settings are moved to the Windows Settings app.

- Win32 apps should in time move over to WinUI Desktop, so those old controls will phase out.

- Hopefully the Windows Shell and inbox apps will move over to WinUI controls, or old apps moved to the store.

**Comment by Poopooracoocoo** [comment]
> Control Panel is due to be removed, once the settings are moved to the Windows Settings app.

One day™

>Win32 apps should in time move over to WinUI Desktop, so those old controls will phase out.

One day™

>Hopefully the Windows Shell and inbox apps will move over to WinUI controls, or old apps moved to the store.

Hopefully®

Microsoft has had so many chances. They've only recently been updating their inbox apps to use WinUI. I don't have any hopes for them moving controls over to Settings from Control Panel. I even made [this ](https://github.com/microsoft/microsoft-ui-xaml/issues/816) issue and the WinUI peeps said that it's out of their hands. It's what I was expecting from Microsoft but part of me hoped® that they could contact the people who develop Windows 10. And phasing out those controls is not such a good idea. Microsoft knows that win32 apps will always be around. They're even using them themselves. They should just update the styling to be inline with WinUI. I wonder why the people who work on Windows 10 don't work on any of the things we really ask for. Are there not enough people, teams or resources?

**Comment by AzAgarampur** [comment]
On the other hand, instead of removing animations altogether, use a different, shorter animation. The current animation of expanding from the top to bottom direction looks somewhat dated and is slow (in my opinion.)

Can a short popup-like animation be used? For example the menu popup menu animation that gtk uses is quick and clean.

**Comment by shaheedmalik** [comment]
> Microsoft has had so many chances. They've only recently been updating their inbox apps to use WinUI. I don't have any hopes for them moving controls over to Settings from Control Panel.

It's like Tablet Mode being terrible the last 5 years. It's all on the Shell team.  I understand if stuff  takes long to to do, but something like Settings has made no real progress since Windows 8.

**Comment by Poopooracoocoo** [comment]
@shaheekmalik yeah they stopped moving settings since 2018. I think it has to do with resources. Microsoft really doesn't care about Windows anymore. The rewritten touch keyboard took so long to get to where it is now and the split keyboard still isn't decent. The rewritten task view is finally performant but still has various issues (including not being able to rearrange). the list goes on and on

**Comment by Poopooracoocoo** [comment]
> On the other hand, instead of removing animations altogether, use a different, shorter animation. The current animation of expanding from the top to bottom direction looks somewhat dated and is slow (in my opinion.)
> 
> Can a short popup-like animation be used? For example the menu popup menu animation that gtk uses is quick and clean.

@AzAgarampur 
I thought GNOME got rid of menu bars. :/ IIRC there's no animation when switching menus. If you're referring to just menus, I totally agree.

**Comment by AzAgarampur** [comment]
@Poopooracoocoo Yeah that's what I meant, my bad. I should have said that.

**Comment by Poopooracoocoo** [comment]
@chigy @StephenLPeters no progress?

**Comment by chigy** [comment]
@Poopooracoocoo , we have no immediate plan to alter behavior here. This animation is tied to MenuFlyout and while we wish to look at the animation holistically, it is not our current immediate priority.
